week_2/pet.rb

```diff
@@ -107,9 +121,21 @@ def self.cost_to_feed(pets, days)
   #   2 => ['clownfish'],
   #   4 => ['python']
   # }
-  #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    result = {}
+    def mapping(hash)
+      hash[@animal_type_id] = []
+    end
+    for animal in pets
+      animal.mapping(result)
+    end
```

Comment:
```suggestion
    for animal in pets
      result[animal.animal_type_id] = []
    end
```
You can also avoid using 2 loops by checking if a key already exist
```ruby
if result[animal.animal_type_id]
   #....
else
  result[animal.animal_type_id] = [animal.name]
end
```